app/src/cose.h

```diff
@@ -20,87 +20,186 @@
 #include <optional>
 #include <qcbor/qcbor.h>
 #include <qcbor/qcbor_spiffy_decode.h>
+#include <span>
 #include <string>
 #include <t_cose/t_cose_sign1_verify.h>
 #include <vector>
 
 namespace scitt::cose
 {
   static constexpr int64_t COSE_HEADER_PARAM_ALG = 1;
+  static constexpr int64_t COSE_HEADER_PARAM_CRIT = 2;
   static constexpr int64_t COSE_HEADER_PARAM_CTY = 3;
   static constexpr int64_t COSE_HEADER_PARAM_KID = 4;
   static constexpr int64_t COSE_HEADER_PARAM_X5CHAIN = 33;
 
+  // Temporary made up profile label, we don't expect claims to contain a
+  // profile header parameter yet, but it's useful to have a label so we can
+  // easily act on the absence of a profile.
+  static constexpr int64_t COSE_HEADER_PARAM_PROFILE = 6861;
+
   // Temporary assignments from draft-birkholz-scitt-architecture
   static constexpr int64_t COSE_HEADER_PARAM_ISSUER = 391;
   static constexpr int64_t COSE_HEADER_PARAM_FEED = 392;
   static constexpr int64_t COSE_HEADER_PARAM_SCITT_RECEIPTS = 394;
 
+  // Notary header parameters.
+  static constexpr const char* NOTARY_HEADER_PARAM_SIGNING_SCHEME =
+    "io.cncf.notary.signingScheme";
+  static constexpr const char* NOTARY_HEADER_PARAM_SIGNING_TIME =
+    "io.cncf.notary.signingTime";
+  static constexpr const char* NOTARY_HEADER_PARAM_AUTHENTIC_SIGNING_TIME =
+    "io.cncf.notary.authenticSigningTime";
+  static constexpr const char* NOTARY_HEADER_PARAM_EXPIRY =
+    "io.cncf.notary.expiry";
+
   struct COSEDecodeError : public std::runtime_error
   {
     COSEDecodeError(const std::string& msg) : std::runtime_error(msg) {}
   };
 
   struct ProtectedHeader
   {
+    // All headers are optional here but optionality will later be validated
+    // according to the COSE profile of the claim.
+    std::optional<std::string> profile;
```

Comment:
I'm not sure I understand why we added this, given it's not being used anywhere (and we infer the profile from the presence of header parameters)

Reply:
Removed it in https://github.com/microsoft/scitt-ccf-ledger/pull/90. - Originally it made the structure of the code more clear but now we've got 3 profiles, after splitting the vanilla case into IETF and X509, I think the structure is clear enough without it.